client/verta/verta/pipeline/_pipeline_step.py

```diff
@@ -0,0 +1,213 @@
+        name: str,
+        model_version: RegisteredModelVersion,
+        predecessors: Optional[
+            List["PipelineStep"]
+        ] = None,  # Optional because it could be the first step with no predecessors
+    ):
+        self._name = name
+        self._model_version = model_version
+        self._predecessors = predecessors or list()
+
+        # avoid the need to pass in connection params when building as local object
+        self._registered_model: Optional[RegisteredModel] = None
+
+        # avoid additional http calls to refresh RMV cache.
+        self._registered_model_id = self._model_version.registered_model_id
+
+    def __repr__(self) -> str:
+        sequence = (
+            "\nPipelineStep:",
+            f"step name: {self.name}",
+        )
+        if self._registered_model:  # don't display on local-only objects`
+            sequence += (f"registered_model: {self._registered_model.name}",)
+        sequence += (
+            f"registered_model_version: {self.model_version.name}",
+            f"registered_model_version_id: {self.model_version.id}",
+            f"predecessors: {[s.name for s in self.predecessors]}",
+        )
+        return "\n".join(sequence)
+
+    @property
+    def model_version(self) -> RegisteredModelVersion:
+        return self._model_version
+
+    @model_version.setter
+    def model_version(self, value) -> None:
+        """Raise a more informative error than the default."""
+        raise AttributeError(
+            "can't set attribute 'model_version'; please use set_model_version()"
+        )
+
+    def set_model_version(self, new_model_version: RegisteredModelVersion) -> None:
+        """
+        Change the registered model version associated with this step.
+
+        Parameters
+        ----------
+        model_version : :class:`~verta.registry.entities.RegisteredModelVersion`
+            Registered model version to use for the step.
+        """
+        if not isinstance(new_model_version, RegisteredModelVersion):
+            raise TypeError(
+                f"model_version must be a RegisteredModelVersion object, not {type(new_model_version)}"
+            )
+        self._model_version = new_model_version
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, value) -> None:
+        """Raise a more informative error than the default."""
+        raise AttributeError("can't set attribute 'name'; please use set_name()")
+
+    def set_name(self, name: str) -> None:
+        """
+        Change the name of this step.
+
+        Parameters
+        ----------
+        new_name : str
+            New name to use for the step.
+        """
+        if not isinstance(name, str):
+            raise TypeError(f"name must be a string, not {type(name)}")
+        self._name = name
+
+    @property
+    def predecessors(self) -> List["PipelineStep"]:
+        return list(set(self._predecessors))  # deduplicated
```

Comment:
I've refactored both predecessors and graph steps to use sets instead of lists, as I think it makes more sense and conveys both uniqueness and irrelevance of ordering.